.github/actions/setup-zeebe/action.yml

```diff
@@ -44,6 +44,10 @@ runs:
     - name: Import Secrets
       id: secrets
       uses: hashicorp/vault-action@v2.4.3
+      if: |
+        ${{ github.event.inputs.secret_vault_address != ''
+        && github.event.inputs.secret_vault_roleId != ''
+        && github.event.inputs.secret_vault_secretId != '' }}
```

Comment:
This condition will only use CI Nexus cache when credentials to Vault are available to the workflow. Those credentials could be unavailable due to 2 reasons:

1. workflow runs for a PR from a fork (intended use case)
2. we misconfigured the Vault credentials so they are missing or wrong

My concern with this condition: in the 2nd case we wouldn't learn about the problem as the workflow would silently swallow the credential error with Vault. Leads to increased cost until eventually the problem is discovered.

Please tell me if my concern is valid or whether there are any other workflows directly using Vault in this repo that would notice problems with the config?
Otherwise, I'd prefer a condition that explicitly checks for `if workflow runs on PR from fork`. See https://github.com/orgs/community/discussions/26829 for code for this.

Reply:
So we should fail if it's not a fork and secrets are not available. We could add that as a failing step with a condition that repo == 'camunda/zeebe' && secrets == null or something along those lines

Reply:
I would definitely like to see some check like this since otherwise we may never detect misconfiguration.

Reply:
I ran it with debug logging enabled: https://github.com/camunda/zeebe/actions/runs/4261817150/jobs/7416704974
```
##[debug]Evaluating: (success() && (github.event.pull_request.head.repo.full_name == 'camunda/zeebe') && ((github.event.inputs.secret_vault_address == '') || (github.event.inputs.secret_vault_roleId == '') || (github.event.inputs.secret_vault_secretId == '')))
##[debug]Evaluating And:
##[debug]..Evaluating success:
##[debug]..=> true
##[debug]..Evaluating Equal:
##[debug]....Evaluating Index:
##[debug]......Evaluating Index:
##[debug]........Evaluating Index:
##[debug]..........Evaluating Index:
##[debug]............Evaluating Index:
##[debug]..............Evaluating github:
##[debug]..............=> Object
##[debug]..............Evaluating String:
##[debug]..............=> 'event'
##[debug]............=> Object
##[debug]............Evaluating String:
##[debug]............=> 'pull_request'
##[debug]..........=> Object
##[debug]..........Evaluating String:
##[debug]..........=> 'head'
##[debug]........=> Object
##[debug]........Evaluating String:
##[debug]........=> 'repo'
##[debug]......=> Object
##[debug]......Evaluating String:
##[debug]......=> 'full_name'
##[debug]....=> 'camunda/zeebe'
##[debug]....Evaluating String:
##[debug]....=> 'camunda/zeebe'
##[debug]..=> true
##[debug]..Evaluating Or:
##[debug]....Evaluating Equal:
##[debug]......Evaluating Index:
##[debug]........Evaluating Index:
##[debug]..........Evaluating Index:
##[debug]............Evaluating github:
##[debug]............=> Object
##[debug]............Evaluating String:
##[debug]............=> 'event'
##[debug]..........=> Object
##[debug]..........Evaluating String:
##[debug]..........=> 'inputs'
##[debug]........=> null
##[debug]......=> null
##[debug]......Evaluating String:
##[debug]......=> ''
##[debug]....=> true
##[debug]..=> true
##[debug]=> true
##[debug]Expanded: (true && ('camunda/zeebe' == 'camunda/zeebe') && ((null == '') || (github['event']['inputs']['secret_vault_roleId'] == '') || (github['event']['inputs']['secret_vault_secretId'] == '')))
##[debug]Result: true
```

Maybe you need to use `inputs.secret_vault_address` instead of `github.event.inputs.secret_vault_address`?

Reply:
Yeah I think that's it:
```
##[debug]..Evaluating Or:
##[debug]....Evaluating Equal:
##[debug]......Evaluating Index:
##[debug]........Evaluating Index:
##[debug]..........Evaluating Index:
##[debug]............Evaluating github:
##[debug]............=> Object
##[debug]............Evaluating String:
##[debug]............=> 'event'
##[debug]..........=> Object
##[debug]..........Evaluating String:
##[debug]..........=> 'inputs'
##[debug]........=> null
```

tells us that `github.event` has no field called `inputs`.

Reply:
Thanks @oleschoenburg, all should be good now 🙂